EMP-408 create start update test script

raysiudak opened the pull request:
Do not merge before EMP-417 is merged.

raysiudak committed:
Add initial start update test script

- This includes the basic components needed to execute
  start_update script tests.
- Add test script execution to build_for_ultimaker
- Add test script execution to CI

Contributes to EMP-389 and EMP-408

Signed-off-by: Raymond Siudak <r.siudak@ultimaker.com>

raysiudak committed:
Add start_update script tests

Sanity check tests:

- test_partition_table_not_found_nok
- test_rsync_ignore_file_not_found_nok
- test_update_rootfs_corrupt_nok
- test_multiple_update_rootfs_files_nok

Execution test:

- test_successful_update_ok

Contributes to EMP-389 and EMP-408

Signed-off-by: Raymond Siudak <r.siudak@ultimaker.com>

oliv3r reviewed (changes requested):
will approve after rebase and all that

oliv3r commented on scripts/start_update.sh:
```diff
@@ -227,12 +227,12 @@ if [ ! -d "${TOOLBOX_MOUNT}${SYSTEM_UPDATE_DIR}" ]; then
 
 if [ ! -f "${TOOLBOX_MOUNT}${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE}.sha512" ]; then
-    echo "Error, update failed: '${TOOLBOX_MOUNT}${TOOLBOX_MOUNT}${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE}.sha512' not found."
+    echo "Error, update failed: '${TOOLBOX_MOUNT}${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE}.sha512' not found."
```
next time, keep slashes between directory variables, this to avoid having errors when variable names are miss-configured (TOOLBOX_MOUNT='/bla/bla' rather then ='/bla/bla/)

oliv3r commented on test/start_update.sh:
```diff
@@ -16,9 +16,10 @@ SYSTEM_UPDATE_DIR="${SYSTEM_UPDATE_DIR:-/etc/system_update}"
 TARGET_STORAGE_DEVICE=""
 
-#SYSTEM_EXECUTABLE_DIR="/sbin"
+SYSTEM_EXECUTABLE_DIR="/sbin"
```
next time, use `SBINDIR="${PREFIX}/sbin}"`
but more importantly, merge this commit into the first commit as you add it there, commented.

oliv3r commented on test/include/chroot_env.sh:
```diff
@@ -29,7 +29,7 @@ setup_chroot_env()
     # in the ro squashfs for testing purposes,
     # e.g. ARM_EMU_BIN mount point and temporary test files.
-    overlayfs_dir="$(mktemp -d -t "${NAME_TEMPLATE_OVERLAYFS}.XXXXXXXX")"
+    overlayfs_dir="$(mktemp -d -t "${NAME_TEMPLATE_OVERLAYFS}.XXXXXX")"
```
how is this still part of this PR? This was fixed in b56df1f7633541014cb2dedf8047e9f5fafe1b9b which is part of master.

oliv3r commented on test/start_update.sh:
```diff
@@ -34,6 +35,10 @@ TEMP_TEST_UPDATE_ROOTFS_FILE="rootfs-v1.2.3.tar.xz"
+NAME_TEMPLATE_TOOLBOX="um-update-toolbox"
+NAME_TEMPLATE_WORKDIR="start_update_workdir"
+NAME_TEMPLATE_UPDATEROOT="temp_update_mount"
+
```
these changes where part of a277c7220e0433029daefa844d880b19bc99382c which is part of master?

oliv3r commented on test/start_update.sh:
```diff
@@ -16,9 +16,10 @@ SYSTEM_UPDATE_DIR="${SYSTEM_UPDATE_DIR:-/etc/system_update}"
-#SYSTEM_EXECUTABLE_DIR="/sbin"
+SYSTEM_EXECUTABLE_DIR="/sbin"
 JEDI_PARTITION_TABLE_FILE_NAME="config/jedi_emmc_sfdisk.table"
-#START_UPDATE_COMMAND="${SYSTEM_EXECUTABLE_DIR}.d/start_update.sh"
```
ditto

oliv3r committed:
test/update_files: Do not test for version files

The ultimaker and debian version file are not part of our test rootfs image, so
lets not test for them.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

oliv3r committed:
Always run chroot teardown first

Ensure that the chroot environment is brought down first. If we do not
do this, we risk bricking the host, due to the increased privileges we
have and have mounted kernel level directories (such as proc and dev)
for example.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>
